test/run_pass/test_require_resolve.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+var assert = require('assert');
+var fixtures = require('../common/fixtures');
+
+assert.strictEqual(
+  require.resolve('./test_util').toLowerCase(),
+  fixtures.path('./test/run_pass/test_util.js').toLowerCase());
+
+assert.strictEqual(
+  require.resolve('os').toLowerCase(), 'os');
```

Comment:
Why should return of require.resolve be lower cased? Is not it expected to be strictly equal to the given strings?

Reply:
Node do this, but my understanding is that sometimes we will get different form by same path, for example:
```shell
$cd ~/GitHub/shadownode
$cd ~/GitHub/ShadowNode
```
these two cmds will enter one same dir., it will be when query the path too, so it is strict to  unify the string form.